contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/MongoGroupScan.java

```diff
@@ -297,30 +279,28 @@ private void handleUnshardedCollection(List<String> hosts) {
     String host = hosts.get(0);
     ServerAddress address = new ServerAddress(host);
     ChunkInfo chunkInfo = new ChunkInfo(hosts, chunkName);
-    chunkInfo.setMinFilters(Collections.<String, Object> emptyMap());
-    chunkInfo.setMaxFilters(Collections.<String, Object> emptyMap());
+    chunkInfo.setMinFilters(Collections.emptyMap());
+    chunkInfo.setMaxFilters(Collections.emptyMap());
     List<ChunkInfo> chunksList = Lists.newArrayList();
     chunksList.add(chunkInfo);
     chunksInverseMapping.put(address.getHost(), chunksList);
   }
 
-  private List<String> getPrimaryShardInfo(MongoClient client) {
+  private List<String> getPrimaryShardInfo() {
     MongoDatabase database = storagePlugin.getClient().getDatabase(CONFIG);
     //Identify the primary shard of the queried database.
     MongoCollection<Document> collection = database.getCollection(DATABASES);
     Bson filter = new Document(ID, this.scanSpec.getDbName());
     Bson projection = new Document(PRIMARY, select);
-    Document document = collection.find(filter).projection(projection).first();
-    Preconditions.checkNotNull(document);
+    Document document = Objects.requireNonNull(collection.find(filter).projection(projection).first());
```

Comment:
An user had reported a problem on our Slack channel (some days ago):
> I am trying to query a Sharded cluster of MongoDB using embedded mode and distributed mode. But both is giving an error of:
query: select * from mongo.db.collection limit 1;
warning   UserRemoteException : 	SYSTEM ERROR: ClassCastException: org.bson.types.ObjectId cannot be cast to java.lang.String
org.apache.drill.common.exceptions.UserRemoteException: SYSTEM ERROR: ClassCastException: org.bson.types.ObjectId cannot be cast to java.lang.String
Please, refer to logs for more information.
[Error Id: 43733103-4cac-4aef-a09f-275df4b4ee4b on ubuntu:31010]
store.mongo.all_text_mode : true
store.mongo.bson.record.reader: false
store.mongo.read_numbers_as_double: true
It throws an error only when collection is sharded, on non-sharded collection, it is working fine.

Could you please check that the failed not an issues? thanks

> Drill 1.18
MongoDB 4.4 Sharded (Not a Replica Sets)

Reply:
I have reproduced it locally and fixed it.